<<<<<<< HEAD
# Enforce `test` and `it` usage conventions (`consistent-test-it`)

ðŸ”§ This rule is automatically fixable by the
[`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->
=======
# Have control over `test` and `it` usages (consistent-test-it)
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294

Jest allows you to choose how you want to define your tests, using the `it` or
the `test` keywords, with multiple permutations for each:

- **it:** `it`, `xit`, `fit`, `it.only`, `it.skip`.
- **test:** `test`, `xtest`, `test.only`, `test.skip`.

<<<<<<< HEAD
## Rule details

This rule gives you control over the usage of these keywords in your codebase.

## Options

This rule can be configured as follows

```json5
{
  type: 'object',
  properties: {
    fn: {
      enum: ['it', 'test'],
    },
    withinDescribe: {
      enum: ['it', 'test'],
    },
  },
  additionalProperties: false,
=======
This rule gives you control over the usage of these keywords in your codebase.

## Rule Details

This rule can be configured as follows

```js
{
    type: 'object',
    properties: {
        fn: {
            enum: ['it', 'test'],
        },
        withinDescribe: {
            enum: ['it', 'test'],
        },
    },
    additionalProperties: false,
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
}
```

#### fn

Decides whether to use `test` or `it`.

#### withinDescribe

<<<<<<< HEAD
Decides whether to use `test` or `it` within a `describe` scope.
=======
Decides whether to use `test` or `it` within a describe scope.
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294

```js
/*eslint jest/consistent-test-it: ["error", {"fn": "test"}]*/

test('foo'); // valid
test.only('foo'); // valid

it('foo'); // invalid
it.only('foo'); // invalid
```

```js
/*eslint jest/consistent-test-it: ["error", {"fn": "it"}]*/

it('foo'); // valid
it.only('foo'); // valid

test('foo'); // invalid
test.only('foo'); // invalid
```

```js
/*eslint jest/consistent-test-it: ["error", {"fn": "it", "withinDescribe": "test"}]*/

it('foo'); // valid
<<<<<<< HEAD
describe('foo', function () {
=======
describe('foo', function() {
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
  test('bar'); // valid
});

test('foo'); // invalid
<<<<<<< HEAD
describe('foo', function () {
=======
describe('foo', function() {
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
  it('bar'); // invalid
});
```

<<<<<<< HEAD
The default configuration forces all top-level tests to use `test` and all tests
nested within `describe` to use `it`.
=======
### Default configuration

The default configuration forces top level test to use `test` and all tests
nested within describe to use `it`.
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294

```js
/*eslint jest/consistent-test-it: ["error"]*/

test('foo'); // valid
<<<<<<< HEAD
describe('foo', function () {
=======
describe('foo', function() {
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
  it('bar'); // valid
});

it('foo'); // invalid
<<<<<<< HEAD
describe('foo', function () {
=======
describe('foo', function() {
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
  test('bar'); // invalid
});
```
