<<<<<<< HEAD
# Disallow using `exports` in files containing tests (`no-export`)

ðŸ’¼ This rule is enabled in the âœ… `recommended`
[config](https://github.com/jest-community/eslint-plugin-jest/blob/main/README.md#shareable-configurations).

<!-- end auto-generated rule header -->

Prevents using `exports` if a file has one or more tests in it.

## Rule details
=======
# no export from test file (no-export)

Prevents exports from test files. If a file has at least 1 test in it, then this
rule will prevent exports.

## Rule Details
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294

This rule aims to eliminate duplicate runs of tests by exporting things from
test files. If you import from a test file, then all the tests in that file will
be run in each imported instance, so bottom line, don't export from a test, but
<<<<<<< HEAD
instead move helper functions into a separate file when they need to be shared
=======
instead move helper functions into a seperate file when they need to be shared
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
across tests.

Examples of **incorrect** code for this rule:

```js
export function myHelper() {}

<<<<<<< HEAD
module.exports = function () {};
=======
module.exports = function() {};
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294

module.exports = {
  something: 'that should be moved to a non-test file',
};

describe('a test', () => {
  expect(1).toBe(1);
});
```

Examples of **correct** code for this rule:

```js
function myHelper() {}

const myThing = {
  something: 'that can live here',
};

describe('a test', () => {
  expect(1).toBe(1);
});
```

## When Not To Use It

Don't use this rule on non-jest test files.
