<<<<<<< HEAD
# Suggest having hooks before any test cases (`prefer-hooks-on-top`)

<!-- end auto-generated rule header -->

While hooks can be setup anywhere in a test file, they are always called in a
specific order, which means it can be confusing if they're intermixed with test
cases.

This rule helps to ensure that hooks are always defined before test cases.

## Rule details
=======
# Suggest to have all hooks at top-level before tests (prefer-hooks-on-top)

All hooks should be defined before the start of the tests

## Rule Details
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294

Examples of **incorrect** code for this rule

```js
/* eslint jest/prefer-hooks-on-top: "error" */

<<<<<<< HEAD
describe('foo', () => {
  beforeEach(() => {
    seedMyDatabase();
  });

  it('accepts this input', () => {
    // ...
  });

  beforeAll(() => {
    createMyDatabase();
  });

  it('returns that value', () => {
    // ...
  });

  describe('when the database has specific values', () => {
    const specificValue = '...';

    beforeEach(() => {
      seedMyDatabase(specificValue);
    });

    it('accepts that input', () => {
      // ...
    });

    it('throws an error', () => {
      // ...
    });

    afterEach(() => {
      clearLogger();
    });
    beforeEach(() => {
      mockLogger();
    });

    it('logs a message', () => {
      // ...
    });
  });

  afterAll(() => {
    removeMyDatabase();
  });
=======
describe("foo" () => {
  beforeEach(() => {
    //some hook code
  });
  test("bar" () => {
    some_fn();
  });
  beforeAll(() => {
    //some hook code
  });
  test("bar" () => {
    some_fn();
  });
});

// Nested describe scenario
describe("foo" () => {
  beforeAll(() => {
    //some hook code
  });
  test("bar" () => {
    some_fn();
  });
  describe("inner_foo" () => {
    beforeEach(() => {
      //some hook code
    });
    test("inner bar" () => {
      some_fn();
    });
    test("inner bar" () => {
      some_fn();
    });
    beforeAll(() => {
      //some hook code
    });
    afterAll(() => {
      //some hook code
    });
    test("inner bar" () => {
      some_fn();
    });
  });
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
});
```

Examples of **correct** code for this rule

```js
/* eslint jest/prefer-hooks-on-top: "error" */

<<<<<<< HEAD
describe('foo', () => {
  beforeAll(() => {
    createMyDatabase();
  });

  beforeEach(() => {
    seedMyDatabase();
  });

  afterAll(() => {
    clearMyDatabase();
  });

  it('accepts this input', () => {
    // ...
  });

  it('returns that value', () => {
    // ...
  });

  describe('when the database has specific values', () => {
    const specificValue = '...';

    beforeEach(() => {
      seedMyDatabase(specificValue);
    });

    beforeEach(() => {
      mockLogger();
    });

    afterEach(() => {
      clearLogger();
    });

    it('accepts that input', () => {
      // ...
    });

    it('throws an error', () => {
      // ...
    });

    it('logs a message', () => {
      // ...
=======
describe("foo" () => {
  beforeEach(() => {
    //some hook code
  });

  // Not affected by rule
  someSetup();

  afterEach(() => {
    //some hook code
  });
  test("bar" () => {
    some_fn();
  });
});

// Nested describe scenario
describe("foo" () => {
  beforeEach(() => {
    //some hook code
  });
  test("bar" () => {
    some_fn();
  });
  describe("inner_foo" () => {
    beforeEach(() => {
      //some hook code
    });
    test("inner bar" () => {
      some_fn();
>>>>>>> 3593505a2789098f8b7e01cec0412530a8c77294
    });
  });
});
```
